rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/IBaseTextDocumentService.java

```diff
@@ -54,6 +54,10 @@ public interface IBaseTextDocumentService extends TextDocumentService {
     void initialized();
     void registerLanguage(LanguageParameter lang);
     void unregisterLanguage(LanguageParameter lang);
+
+    void projectAdded(String name, String uri);
+    void projectRemoved(String name, String uri);
```

Comment:
how about we turn the `String uri` into an `URI` or a `ISourceLocation`? to avoid people getting creative with them in ways we don't want?

Reply:
I totally agree. `ISourceLocations` is fine here and matches some of the other methods in this class.